src/desert/_fields.py

```diff
@@ -0,0 +1,84 @@
+import typing
+
+import attr
+import marshmallow.fields
+
+
+T = typing.TypeVar("T")
+
+
+@attr.s(frozen=True, auto_attribs=True)
+class TypeTagField:
+    cls: type
+    tag: str
+    field: typing.Type[marshmallow.fields.Field]
+
+
+@attr.s(auto_attribs=True)
+class TypeDictFieldRegistry:
+    the_dict: typing.Dict[typing.Union[type, str], marshmallow.fields.Field] = attr.ib(
```

Comment:
I'm not a big `the_foo` user, perhaps `.mapping` or something.

Reply:
Heh, yeah, that was a garbage name so as to not think about it and worry about other things.  Could also just be `.dict`